dmutils/repoutils/freeze_requirements.py

```diff
@@ -72,12 +73,14 @@ def exit_failure(messages, code=1):
         exit_failure([install_cmd.stdout, install_cmd.stderr])
 
     logger.info(f'Freezing installed requirements in virtualenv: {virtualenv_name}')
-    freeze_cmd = subprocess.run(
-        [f'{virtualenv_name}/bin/pip', 'freeze', '-r', target],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        universal_newlines=True
-    )
+    with open(target, 'rb') as requirements_file, tempfile.NamedTemporaryFile() as tmp:
+        tmp.write(re.sub(b"--.+?\n", b"", requirements_file.read()))
+        freeze_cmd = subprocess.run(
+            [f'{virtualenv_name}/bin/pip', 'freeze', '-r', f'{tmp.name}'],
```

Comment:
As an alternative to creating a tempfile you could use `shell=True` in the arguments to subprocess.run; then the context manager would be removed and the command to subprocess would be the same as in the Makefile:

```suggestion
            f"{virtualenv_name}/bin/pip freeze -r <(sed '/^--/d' {target})"],
            shell=True,
```

The functionality would be exactly the same, real question is which is more readable.

Reply:
I wanted to do as much as possible in python, including the replace. I think part of the reason I made this mistake in the first place is my unfamiliarity with bash so more python > more bash for me.

I've also heard it's best to avoid `shell=True` although I'm not 100% sure I understand why.

Reply:
Awesome, leave it as it is then 👍